### ISSUE TITLE
Clang-cl treats -Wall as -Weverything

### ISSUE THREAD

**Issue by davidavetyan** [llvm/llvm-project#102583]
When passing the `-Wall` option to clang-cl options, all diagnostics are being reported. This would be the expected behavior for `/Wall` which enables `-Weverything`, but not for `-Wall`.

From the clang-cl user manual https://clang.llvm.org/docs/UsersManual.html#clang-cl:
```
  /Wall                   Enable -Weverything
  ...
  -W<warning>             Enable the specified warning
```

Used LLVM version 17.0.6

**Comment by adalisk-emikhaylov** [comment]
I believe this is intended, because `cl.exe` itself accepts options with both `/` and `-`,

**Comment by davidavetyan** [comment]
Ok, so I guess this is required for compatibility.
It'd be good to update the documentation to be clear about this, because the other warnings work as usual with '-'.

**Comment by efriedma-quic** [comment]
Yes, this is the way MSVC-style command-line parsing works.  If you want the clang -Wall, you can pass "/clang:-Wall"
